tests/lru.rs

```diff
@@ -0,0 +1,68 @@
+//! Test setting LRU actually limits the number of things in the database;
+use std::sync::{
+    atomic::{AtomicUsize, Ordering},
+    Arc,
+};
+
+use salsa::Database as _;
+
+#[derive(Debug, PartialEq, Eq)]
+struct HotPotato(u32);
+
+static N_POTATOES: AtomicUsize = AtomicUsize::new(0);
+
+impl HotPotato {
+    fn new(id: u32) -> HotPotato {
+        N_POTATOES.fetch_add(1, Ordering::SeqCst);
+        HotPotato(id)
+    }
+}
+
+impl Drop for HotPotato {
+    fn drop(&mut self) {
+        N_POTATOES.fetch_sub(1, Ordering::SeqCst);
+    }
+}
+
+#[salsa::query_group(QueryGroupStorage)]
+trait QueryGroup {
+    fn get(&self, x: u32) -> Arc<HotPotato>;
+}
+
+fn get(_db: &impl QueryGroup, x: u32) -> Arc<HotPotato> {
+    Arc::new(HotPotato::new(x))
+}
+
+#[salsa::database(QueryGroupStorage)]
+#[derive(Default)]
+struct Database {
+    runtime: salsa::Runtime<Database>,
+}
+
+impl salsa::Database for Database {
+    fn salsa_runtime(&self) -> &salsa::Runtime<Database> {
+        &self.runtime
+    }
+}
+
+#[test]
```

Comment:
If it's not too much trouble, can we add a test that modifies `set_lru_capacity` after the fact, too? (Ideally, one to some intermediate value (e.g., 22), one to 0, and one to something bigger (64))